src/client-ws/ws.js

```diff
@@ -123,54 +79,134 @@ class WSClient extends EventTarget {
       const message = JSON.parse(chunk);
       if (Array.isArray(message)) {
         // possible batch request
-        this.handleBatchResponse(message);
+        message.forEach((res) => {
+          if (!res.id) {
```

Comment:
This will throw an error if the client receieves an empty array, or invalid array.